cpp/common/src/codingstandards/cpp/rules/identifierhidden/IdentifierHidden.qll

```diff
@@ -11,13 +11,36 @@ abstract class IdentifierHiddenSharedQuery extends Query { }
 
 Query getQuery() { result instanceof IdentifierHiddenSharedQuery }
 
-query predicate problems(UserVariable v2, string message, UserVariable v1, string varName) {
+/**
+ * There is a lambda that contains a declaration
+ * that hides something that is captured
+ * and the lambda exists in the function where this lamda is enclosed
+ */
+predicate hiddenInLambda(UserDeclaration v2, UserDeclaration v1) {
```

Comment:
To make it consistent with `hidesStrict` we should swap the parameters.
`v2` hides `v1`.

```suggestion
predicate hiddenInLambda(UserDeclaration v1, UserDeclaration v2) {
```

Reply:
I think overall using of more verbose variables help with understanding the predicate.

```ql
predicate hiddenInLambda(UserDeclaration outerDecl, UserDeclaration innerDecl) {
  exists(Scope scopeOfInnerDecl, Closure closureWithInnerDecl|
    scopeOfInnerDecl.getADeclaration() = innerDecl and
    scopeOfInnerDecl.getAnAncestor() = closureWithInnerDecl and
    closureWithInnerDecl.getEnclosingFunction().getBasicBlock().(Scope) = outerDecl.getParentScope() and
    exists(LambdaCapture cap, Variable v |
      v.getAnAccess() = cap.getInitializer().(VariableAccess) and
      v = outerDecl and
      closureWithInnerDecl.getLambdaExpression().getACapture() = cap
    ) and
    innerDecl.getName() = outerDecl.getName()
  )
}
```

Reply:
Since we are limiting to variable in the `exists` it would be a performance benefit to change the parameters to `UserVariable` to potentially reduce the number of starting tuples.

Reply:
the scope check here is to say - the innerDecl is in a scope inside the lambda, its not for the capture (but this is hopefully more clear now with the var name improvements 😅 )

not sure I understand the case - "The situation becomes more tricky for variables that accessible without a capture." 
if the variable is not captured, then there is no hiding, reference to the compliant exception example in the standard

Reply:
A lambda, can for example refer to a global variable without capturing that.

```
int foo = 0;

int bar() {
    auto lambda_with_shadowing = []() {
        int foo = 1;
        return foo + foo;
    };

    auto lambda_without_shadowing = []() {
        return foo + foo;
    };

    return lambda_with_shadowing();
}
```

I would argue that `lambda_with_shadowing` is non compliant.

Reply:
aha, you are correct ,

I have added cases for the description [here](https://en.cppreference.com/w/cpp/language/lambda) when a lambda can access a variable without explicitly capturing it